sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,143 @@
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_POSTGRESQL_AUTHENTICATION_PLUGIN_CLASSNAME;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_NAME_POSTGRESQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_VALUE_MYSQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_VALUE_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.implementation.jdbc.JdbcPropertyConstants.PROPERTY_VALUE_POSTGRESQL_SSL_MODE;
+
+/**
+ */
+class JdbcPropertiesBeanPostProcessor implements BeanPostProcessor, EnvironmentAware {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcPropertiesBeanPostProcessor.class);
+    private static final String SPRING_TOKEN_CREDENTIAL_PROVIDER = "com.azure.spring.cloud.service.implementation.identity.impl.credential.provider.SpringTokenCredentialProvider";
+
+    private final AzureGlobalProperties azureGlobalProperties;
+
+    private Environment environment;
+
+    public JdbcPropertiesBeanPostProcessor(AzureGlobalProperties azureGlobalProperties) {
+        this.azureGlobalProperties = azureGlobalProperties;
+    }
+
+    private static final Map<String, String> POSTGRES_ENHANCED_PROPERTIES = new TreeMap<>();
+    private static final Map<String, String> MYSQL_ENHANCED_PROPERTIES = new TreeMap<>();
+    static final Map<DatabaseType, Map<String, String>> DEFAULT_ENHANCED_PROPERTIES = new HashMap<>();
+
+    static {
+        POSTGRES_ENHANCED_PROPERTIES.put(PROPERTY_NAME_POSTGRESQL_AUTHENTICATION_PLUGIN_CLASSNAME, POSTGRES_AUTH_PLUGIN_CLASS_NAME);
+        POSTGRES_ENHANCED_PROPERTIES.put(PROPERTY_NAME_POSTGRESQL_SSL_MODE, PROPERTY_VALUE_POSTGRESQL_SSL_MODE);
+
+        MYSQL_ENHANCED_PROPERTIES.put(PROPERTY_NAME_MYSQL_SSL_MODE, PROPERTY_VALUE_MYSQL_SSL_MODE);
+        MYSQL_ENHANCED_PROPERTIES.put(PROPERTY_NAME_MYSQL_USE_SSL, PROPERTY_VALUE_MYSQL_USE_SSL);
+        MYSQL_ENHANCED_PROPERTIES.put(PROPERTY_NAME_MYSQL_DEFAULT_AUTHENTICATION_PLUGIN, MYSQL_AUTH_PLUGIN_CLASS_NAME);
+        MYSQL_ENHANCED_PROPERTIES.put(PROPERTY_NAME_MYSQL_AUTHENTICATION_PLUGINS, MYSQL_AUTH_PLUGIN_CLASS_NAME);
+
+        DEFAULT_ENHANCED_PROPERTIES.put(DatabaseType.MYSQL, MYSQL_ENHANCED_PROPERTIES);
+        DEFAULT_ENHANCED_PROPERTIES.put(DatabaseType.POSTGRESQL, POSTGRES_ENHANCED_PROPERTIES);
+    }
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        if (bean instanceof DataSourceProperties) {
+            DataSourceProperties dataSourceProperties = Binder.get(environment).bindOrCreate("spring.datasource", DataSourceProperties.class);
```

Comment:
do we need to bind again? It's already a `DataSourceProperties` bean.

Reply:
No, updated.